### DISCUSSION TITLE
Can you give me some pointers on how to setup this mono-repo.

### DISCUSSION THREAD

**Discussion by Carlosg19** [dart-lang/i18n#902] [Q&A] [answered]
The repo is labeled as a mono-repo. But, it's no making use of any mono-repo management tool such as mono_repo, or Melos. I am confused. I would really appreciate if someone can give me some pointers.

**Comment by mosuem** [comment, marked as answer]
Thanks for using this new discussions feature :)

I personally just use it as a normal repo - my workflow is 

```
git clone ...
cd pkgs/messages
dart pub get
dart run ...
```

I do want to switch to use the new pub workspace feature, that should help even more. Any suggestions or help are appreciated, I am curious to what workflows are available. I have no experience with `melos`, only `mono_repo`, which I find adds complexity more than it removes it.

**Reply by Carlosg19** [reply]
Thanks. I actually just did the same thing you did. For the sake of consistency. Lol. I am not really a fan of mono-repo. But, in this type a situations. I think setting up Melos would simplify things for all the current and future contributors.

**Reply by mosuem** [reply]
Maybe I should take a look at `melos`... What are current problems this might solve?

**Reply by Carlosg19** [reply]
it's pretty much just creates a single "source of truth" to setup the project. You clone the repo, then write some specific commands specified in the readme and done. you are ready to contribute.

It doesn't do anything to the technical aspects project. The project is not even a huge thing. It just makes the setup smoother, and more standard among contributors. It only improves the DX.
